sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java

```diff
@@ -552,7 +557,19 @@ public void serialize(PipelineOptions value, JsonGenerator jgen, SerializerProvi
         jgen.writeObject(serializableOptions);
 
         List<Map<String, Object>> serializedDisplayData = Lists.newArrayList();
-        for (DisplayData.Item<?> item : DisplayData.from(value).items()) {
+        DisplayData displayData;
+
+        try {
+          displayData = DisplayData.from(value);
+        } catch (Exception e) {
+          String msg = "Exception thrown while collecting display data for PipelineOptions. "
```

Comment:
This is for the entire set of `PipelineOptions` values, we serialize and collect all `DisplayData` as a set.

We do iterate over each property inside `ProxyInvocationHandler.populateDisplayData`, where we could catch-and-rethrow exceptions with extra context of the property which caused the exception. However we don't run any user-code in this loop so I don't think it's worth complicating the code.